### DISCUSSION TITLE
Which one is better for Symfony 5.3.6? PHP 7.4 or PHP 8.0.9 ?

### DISCUSSION THREAD

**Discussion by karaka200** [symfony/symfony#42527] [Q&A] [answered]
I'm new to the Symfony.
Which one is better for Symfony 5.3.6?
PHP 7.4 or PHP 8.0.9 ?
Please explain to me the reason.

**Comment by faizanakram99** [comment, marked as answer]
It is always better to use latest PHP versions in order to use latest language features.

PHP 8 provides native union types, native attributes, match operator, constructor promotion and more. These shiny new features increase overall Developer experience and also help in writing better code. I see no reason to use an old PHP version when u can use the latest version.

You can read more about it here
https://stitcher.io/blog/new-in-php-8


Supporting old versions of PHP is a necessity for libraries and not for end user projects.

By the way, Symfony 5.3 works with any PHP version greater than or equal to 7.2.5.

**Reply by abzolv** [reply]
It's not quite as clear-cut. The highest PHP version you can use is determined by the highest PHP version that is supported by all the third-party packages that you're using.

**Reply by ahundiak** [reply]
I would argue that if a third-party package does not run under 8.0 then you probably should not be using it.  But I get your point.

To me a more important blocking point is which PHP version does your deployment strategy require/support.
